libs/sync/path.go

```diff
@@ -43,27 +44,20 @@ func EnsureRemotePathIsUsable(ctx context.Context, wsc *databricks.WorkspaceClie
 	if err != nil {
 		// We only deal with 404s below.
 		if !apierr.IsMissing(err) {
-			return err
+			return false, err
 		}
 
 		// If the path is nested under a repo, the repo has to exist.
 		if strings.HasPrefix(remotePath, "/Repos/") {
 			repoPath := repoPathForPath(me, remotePath)
 			_, err = wsc.Workspace.GetStatusByPath(ctx, repoPath)
 			if err != nil && apierr.IsMissing(err) {
-				return fmt.Errorf("%s does not exist; please create it first", repoPath)
+				return false, fmt.Errorf("%s does not exist; please create it first", repoPath)
 			}
 		}
 
-		// The workspace path doesn't exist. Create it and try again.
-		err = wsc.Workspace.MkdirsByPath(ctx, remotePath)
-		if err != nil {
-			return fmt.Errorf("unable to create directory at %s: %w", remotePath, err)
-		}
-		info, err = wsc.Workspace.GetStatusByPath(ctx, remotePath)
-		if err != nil {
-			return err
-		}
+		return false, nil
+
```

Comment:
Can we still test if we have permission to write to a parent that does exist?

We run this during validation, so if we cannot write to the destination path, it's good to surface early.